pages/app/ExploreDetail.js

```diff
@@ -155,35 +154,17 @@ class ExploreDetail extends Page {
   }
 
   getSimilarDatasets() {
-    this.setState({
-      similarDatasetsLoaded: false
-    });
+    this.setState({ similarDatasetsLoaded: false });
+
     this.datasetService.getSimilarDatasets()
-      .then((response) => {
-        let counter = 0;
-        const similarDatasets = response.map(val => val.dataset).filter(
-          () => {
-            counter++;
-            return counter < 7;
-          });
-
-        if (similarDatasets.length > 0) {
-          DatasetService.getDatasets(similarDatasets, 'widget,metadata,layer,vocabulary')
-            .then((data) => {
-              this.setState({
-                similarDatasetsLoaded: true,
-                similarDatasets: data
-              });
-            })
-            .catch(err => toastr.error('Error', err));
-        } else {
-          this.setState({
-            similarDatasetsLoaded: true,
-            similarDatasets: []
-          });
-        }
+      .then(res => res.map(val => val.dataset).slice(0, 7))
+      .then((ids) => {
+        if (ids.length === 0) return [];
+        return DatasetService.getDatasets(ids, 'widget,metadata,layer,vocabulary');
       })
-      .catch(err => toastr.error('Error', err));
+      .then(similarDatasets => this.setState({ similarDatasets }))
+      .catch(() => toastr.error('Error', 'Unable to load the similar datasets'))
```

Comment:
Could you also log the err object to the error console here?

Reply:
Yes! Actually that was one of my doubts. Don't you think a `console.error` is better than a `console.log`?